tests.py

```diff
@@ -14,6 +14,28 @@
 
 
 if __name__ == '__main__':
+
+    try:
+        import setuptools
+    except ImportError:
+        pass
+
+    # The warnings need to be before any of matplotlib imports, but after
+    # setuptools (if present) which has syntax error with the warnings enabled.
+    # Filtering by module does not work as this will be raised by Python itself.
+    # so `module=matplotlib.*` is out of questions.
+
+    import warnings
+
+    # Python 3.6 deprecate invalid character-pairs \A, \* ... in non raw-strings
+    # and other things. Let's not re-introduce them
+    warnings.filterwarnings('error', '.*invalid escape sequence.*',
+        category=DeprecationWarning)
+    warnings.filterwarnings(
+        'default',
+        '.*inspect.getargspec\(\) is deprecated.*',
+        category=DeprecationWarning)
+
```

Comment:
This only gets executed when running tests.py, and I think most tests have been migrated to pytest now. Is there a way to make pytest execute these filterwarnings calls?